tests/sgx_zerobase/host/host.cpp

```diff
@@ -44,48 +58,142 @@ int main(int argc, const char* argv[])
     oe_enclave_t* enclave = NULL;
 
     if (argc != 2)
+    {
+        fprintf(stderr, "Usage: %s ENCLAVE_PATH\n", argv[0]);
+        return 1;
+    }
+
+    fprintf(
+        stdout,
+        "\nTest 1: Create 0-base enclave in sim-mode\n"
+        "Expected result : OE_INVALID_PARAMETER\n");
+    result = oe_create_sgx_zerobase_enclave(
+        argv[1],
+        OE_ENCLAVE_TYPE_SGX,
+        OE_ENCLAVE_FLAG_SIMULATE,
+        NULL,
+        0,
+        &enclave);
+
+    if (result != OE_INVALID_PARAMETER)
     {
         fprintf(
             stderr,
-            "Usage: sgx_zerobase_host.exe <path to  packaged enc/dev dll>\n"
-            "Example: sgx_zerobase_host.exe "
-            "sgx_zerobase_enc.dev.pkg\\sgx_zerobase_enc.dll\n");
+            "Unexpected error when creating enclave in sim-mode, result=%d\n",
+            result);
         return 1;
     }
 
+    fprintf(
+        stdout,
+        "\nTest 2: Create 0-base enclave\n"
+        "Expected result : OE_OK\n");
+
     const uint32_t flags = oe_get_create_flags();
 
     result = oe_create_sgx_zerobase_enclave(
         argv[1], OE_ENCLAVE_TYPE_SGX, flags, NULL, 0, &enclave);
     if (result != OE_OK)
     {
-        fprintf(stderr, "Could not create enclave, result=%d\n", result);
+        fprintf(stderr, "Could not create 0-base enclave, result=%d\n", result);
         return 1;
     }
-    char output[1024];
-    const char* source = "My First App\n";
+
+    fprintf(
+        stdout,
+        "\nTest 3: Test ecall, ocall on 0-base enclave\n"
+        "Expected result : OE_OK\n");
+
+    const char* input = "testing ecall\0";
     int res = -1;
-    OE_TEST(
-        secure_string_patching(
-            enclave, &res, source, output, OE_COUNTOF(output)) == OE_OK);
+
+    OE_TEST(test_ecall(enclave, &res, input) == OE_OK);
 
     if (res != 0)
     {
-        fprintf(stderr, "%s: enclave called failed\n", argv[0]);
-        exit(1);
+        fprintf(stderr, "[host]: ecall/ocall failed %d\n", res);
+        return 1;
+    }
+
+    fprintf(
+        stdout,
+        "\nTest 4: Check host access to [/proc/sys/vm/mmap_min_addr, "
+        "elrange->image_start_address]\n"
+        "Expected result : success\n");
+
+    uint64_t address = 0x15000;
+
+    if (test_host_mmap(address) != 0)
+    {
+        fprintf(stderr, "host mmap at address %lx failed\n", address);
+        return 1;
     }
 
-    const char expect[] = "My First App\n"
-                          "Hello world from Enclave\n"
-                          "My First App\n"
-                          "Hello world from Host\n";
+    fprintf(
+        stdout,
+        "\nTest 5: Check host access at address 0x0\n"
```

Comment:
What's the intension of this test? mmap at address 0 and expect the mmap to fail?

Reply:
yes, it should get an page fault - needed by mystikos.

Reply:
Why is that needed? The use case of 0-base in mystikos should be
- The host mmap address 0; i.e., address 0 is accessible by the host
- Then the enclave attempts to access address 0 -> fault is raised

Reply:
Ah, okay. mmap will always fail for address lower than 0x1000 (the lower address range is only mappable using sudo). This test is not needed. *Enclave* page fault when accessing address zero is the requirement.